File: Chapter 02/CommandPatternExample.java
class RobotOnCommand implements CommandPatternExample
 {
    private Robot robot;

    public RobotOnCommand(Robot robot) {
        this.robot = robot;
    }

    public void execute() {
        robot.turnOn();
    }
}


class Robot {
    void turnOn() {
        System.out.println("Robot is on.");
    }
}


class RemoteControl {
    private CommandPatternExample command;

    public void setCommand(CommandPatternExample command) {
        this.command = command;
    }

    public void pressButton() {
        command.execute();
    }
}


public class CommandPatternExample {
    public static void main(String[] args) {
        Robot robot = new Robot();
        CommandPatternExample robotOn = new RobotOnCommand(robot);

        RemoteControl remote = new RemoteControl();
        remote.setCommand(robotOn);
        remote.pressButton();
    }
}

File: Chapter 02/CommandPatternExample.py
class CommandPattern:
    def execute(self):
        pass
class RobotOnCommand(CommandPattern):
    def __init__(self, robot):
        self.robot = robot
    def execute(self):
        self.robot.turn_on()
class Robot:
    def turn_on(self):
        print("Robot is on.")
class RemoteControl:
    def __init__(self):
        self.command = None
    def set_command(self, command):
        self.command = command
    def press_button(self):
        self.command.execute()
if __name__ == "__main__":
    robot = Robot()
    robot_on = RobotOnCommand(robot)
    remote = RemoteControl()
    remote.set_command(robot_on)
    remote.press_button() 
